packages/skeleton/src/components/project.ts

```diff
@@ -0,0 +1,462 @@
+      this.executeAdditionalChecks()
+    }
+  }
+
+  executeAdditionalChecks(): void {
+    this.cells?.forEach((cell) => {
+      cell.executeAdditionalChecks()
+    })
+    this.directives?.forEach((directive) => {
+      directive.executeAdditionalChecks()
+    })
+    this.functions?.forEach((func) => {
+      func.executeAdditionalChecks()
+    })
+    this.layouts?.forEach((layout) => {
+      layout.executeAdditionalChecks()
+    })
+    this.pages?.forEach((page) => {
+      page.executeAdditionalChecks()
+    })
+    this.routers?.forEach((router) => {
+      router.executeAdditionalChecks()
+    })
+    this.sdls?.forEach((sdl) => {
+      sdl.executeAdditionalChecks()
+    })
+    this.services?.forEach((service) => {
+      service.executeAdditionalChecks()
+    })
+    this.sides?.forEach((side) => {
+      side.executeAdditionalChecks()
+    })
+    this.tomls?.forEach((toml) => {
+      toml.executeAdditionalChecks()
+    })
+  }
+
+  getTOMLs(forceExtract = false): RedwoodTOML[] {
+    if (forceExtract || this.tomls === undefined) {
+      this.tomls = extractTOMLs(this)
+    }
+    return this.tomls
+  }
+
+  getSDLs(forceExtract = false): RedwoodSDL[] {
+    if (forceExtract || this.sdls === undefined) {
+      this.sdls = extractSDLs(this)
+    }
+    return this.sdls
+  }
+
+  getServices(forceExtract = false): RedwoodService[] {
+    if (forceExtract || this.services === undefined) {
+      this.services = extractServices(this)
+    }
+    return this.services
+  }
+
+  getSides(forceExtract = false): RedwoodSide[] {
+    if (forceExtract || this.sides === undefined) {
+      this.sides = extractSides(this)
+    }
+    return this.sides
+  }
+
+  getCells(forceExtract = false): RedwoodCell[] {
+    if (forceExtract || this.cells === undefined) {
+      this.cells = extractCells(this)
+    }
+    return this.cells
+  }
+
+  getRouters(forceExtract = false): RedwoodRouter[] {
+    if (forceExtract || this.routers === undefined) {
+      this.routers = extractRouters(this)
+    }
+    return this.routers
+  }
+
+  getLayouts(forceExtract = false): RedwoodLayout[] {
+    if (forceExtract || this.layouts === undefined) {
+      this.layouts = extractLayouts(this)
+    }
+    return this.layouts
+  }
+
+  getPages(forceExtract = false): RedwoodPage[] {
+    if (forceExtract || this.pages === undefined) {
+      this.pages = extractPages(this)
+    }
+    return this.pages
+  }
+
+  getFunctions(forceExtract = false): RedwoodFunction[] {
+    if (forceExtract || this.functions === undefined) {
+      this.functions = extractFunctions(this)
+    }
+    return this.functions
+  }
+
+  getDirectives(forceExtract = false): RedwoodDirective[] {
+    if (forceExtract || this.directives === undefined) {
+      this.directives = extractDirectives(this)
+    }
+    return this.directives
+  }
+
+  hasWarnings(cascade = false): boolean {
+    let warningsFound = this.warnings.length > 0
```

Comment:
May as well use the class you're extending from:

```suggestion
    let warningsFound = super.hasWarnings()
```